internal/transformer/transformations/transformations.go

```diff
@@ -18,194 +18,112 @@ package transformations
 
 import (
 	"fmt"
+	"strings"
 
 	"github.com/konveyor/move2kube/internal/common"
 	"github.com/konveyor/move2kube/internal/qaengine"
 	"github.com/konveyor/move2kube/internal/starlark/gettransformdata"
 	"github.com/konveyor/move2kube/internal/starlark/types"
+	qatypes "github.com/konveyor/move2kube/types/qaengine"
 	log "github.com/sirupsen/logrus"
-	"github.com/spf13/cast"
-)
-
-var (
-	questions = map[string]types.MapT{}
 )
 
 // Question answer functions
 
-// AskStaticQuestion asks a static question
-func AskStaticQuestion(questionObjI interface{}) error {
-	log.Trace("start myStaticAskQuestion")
-	defer log.Trace("end myStaticAskQuestion")
-	questionObj, ok := questionObjI.(types.MapT)
-	if !ok {
-		return fmt.Errorf("Expected questions to be of map type. Actual value is %+v of type %T", questionObjI, questionObjI)
-	}
-	qakeyI, ok := questionObj["key"]
-	if !ok {
-		return fmt.Errorf("The key 'key' is missing from the question object %+v", questionObj)
-	}
-	qakey, ok := qakeyI.(string)
-	if !ok {
-		return fmt.Errorf("The key 'key' is not a string. The question object %+v", questionObj)
-	}
-	if _, ok := questionObj["type"]; !ok {
-		questionObj["type"] = "string"
-	}
-	questions[qakey] = questionObj
-	return nil
-}
-
-// AnswerFn answers a static question
-func AnswerFn(qakey string) (interface{}, error) {
-	log.Trace("start myAnswerFn")
-	defer log.Trace("end myAnswerFn")
-	questionObj, ok := questions[qakey]
-	if !ok {
-		return nil, fmt.Errorf("There is no question with the key: %s", qakey)
-	}
-	return askQuestion(questionObj)
-}
-
 // AskDynamicQuestion asks a dynamic question
 func AskDynamicQuestion(questionObjI interface{}) (interface{}, error) {
+	log.Trace("start AskDynamicQuestion")
+	defer log.Trace("end AskDynamicQuestion")
+
 	questionObj, ok := questionObjI.(types.MapT)
 	if !ok {
-		return nil, fmt.Errorf("Expected question to be of map type. Actual value is %+v of type %T", questionObjI, questionObjI)
-	}
-	if _, ok := questionObj["type"]; !ok {
-		questionObj["type"] = "string"
+		return nil, fmt.Errorf("expected question to be of map type. Actual value is %+v of type %T", questionObjI, questionObjI)
 	}
 	return askQuestion(questionObj)
 }
 
-func askQuestion(questionObj types.MapT) (interface{}, error) {
-	log.Trace("start commonStuff")
-	defer log.Trace("end commonStuff")
+func convertMapTToProblem(questionObj types.MapT) (qatypes.Problem, error) {
+	log.Trace("start convertMapTToProblem")
+	defer log.Trace("end convertMapTToProblem")
+
+	prob := qatypes.Problem{}
+
+	// key
 	qakeyI, ok := questionObj["key"]
```

Comment:
Aren't we going to use unmarshal to do this, so that we can keep the structure in all interfaces consistent?

Reply:
done